group-coordinator/src/main/java/org/apache/kafka/coordinator/group/runtime/CoordinatorLoader.java

```diff
@@ -26,7 +27,40 @@
  *
  * @param <U> The type of the record.
  */
-public interface CoordinatorLoader<U> {
+public interface CoordinatorLoader<U> extends AutoCloseable {
+
+    /**
+     * UnknownRecordTypeException is thrown when the Deserializer encounters
+     * an unknown record type.
+     */
+    class UnknownRecordTypeException extends RuntimeException {
+        private final short unknownType;
+
+        public UnknownRecordTypeException(short unknownType) {
+            super(String.format("Found an unknown record type %d", unknownType));
+            this.unknownType = unknownType;
+        }
+
+        public short unknownType() {
+            return unknownType;
```

Comment:
Do we use this anywhere?

Reply:
Ah I see in the tests.

Reply:
It is also used in CoordinatorLoadedImpl.

Reply:
ah in the log message 👍